internal/cmd/oohelperd/internal/nwcth/http.go

```diff
@@ -0,0 +1,110 @@
+package nwcth
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+
+	"github.com/ooni/probe-cli/v3/internal/iox"
+	"github.com/ooni/probe-cli/v3/internal/runtimex"
+)
+
+// HTTPConfig configures the HTTP check.
+type HTTPConfig struct {
+	Jar       *cookiejar.Jar
+	Headers   map[string][]string
+	Transport http.RoundTripper
+	URL       *url.URL
+}
+
+// HTTPDo performs the HTTP check.
+func HTTPDo(ctx context.Context, config *HTTPConfig) (*CtrlHTTPRequest, *NextLocationInfo) {
+	req, err := newRequest(ctx, config.URL)
+	if err != nil {
+		return &CtrlHTTPRequest{Failure: newfailure(err)}, nil
+	}
+	// The original test helper failed with extra headers while here
+	// we're implementing (for now?) a more liberal approach.
+	for k, vs := range config.Headers {
+		switch strings.ToLower(k) {
+		case "user-agent":
+		case "accept":
+		case "accept-language":
+			for _, v := range vs {
+				req.Header.Add(k, v)
+			}
+		}
+	}
+	// redirectReq is a clone of the initial request, with possibly modified headers
+	var redirectReq *http.Request
+
+	jar := config.Jar
+	if jar == nil {
+		jar, err = cookiejar.New(nil)
+		runtimex.PanicOnError(err, "cookiejar.New failed")
+	}
+	client := http.Client{
+		CheckRedirect: func(r *http.Request, reqs []*http.Request) error {
+			redirectReq = r
+			return http.ErrUseLastResponse
+		},
+		Jar:       jar,
+		Transport: config.Transport,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return &CtrlHTTPRequest{Failure: newfailure(err)}, nil
+	}
+	var httpRedirect *NextLocationInfo = nil
```

Comment:
```suggestion
	var httpRedirect *NextLocationInfo
```

Idiomatic ^^^. Stuff is zero initialized in Go.